src/traits.rs

```diff
@@ -0,0 +1,35 @@
+use crate::{evaluation, providers::traits::FeatureProvider, ClientMetaData, EvaluationDetails};
+
+pub trait ClientTraits<C>
+where
+    C: FeatureProvider,
+{
+    fn new(name: String, provider: C) -> Self;
+    fn meta_data(&self) -> ClientMetaData;
+    fn set_evaluation_context(&mut self, eval_ctx: evaluation::EvaluationContext);
+    fn evaluation_context(&self) -> evaluation::EvaluationContext;
+    fn evaluate<T>(
```

Comment:
What's the difference between `evaluate<T>` and `value_details<T>`? The interfaces(?) look identical.

Reply:
Looking at the go client https://github.com/open-feature/go-sdk/blob/main/pkg/openfeature/client.go 

The evaluationdetails gives a different return type, I also made this more confusing I think, I don't even see `evaluate` as a method, that's on the provider, so ill drop it from the client